docs/src/ref_impl/installer.md

```diff
@@ -5,67 +5,56 @@
 
 # Installer
 
-## Configuring and Building Installer for Ghaf
-
-You can obtain the installation image for your Ghaf configuration. To check possible configuration options, see [Modules Options](../ref_impl/modules_options.md#ghafinstallerenable).
-
-1. Set `ghaf.installer.enable` to `true`.
-2. Add nixos-generators module to `ghaf.installer.imgModules` list to configure installer image type.
-3. Choose installer modules from `ghaf.installer.installerModules` and set `ghaf.installer.enabledModules` to list of their names.
-4. Write code for the installer in `ghaf.installer.installerCode`.
-
-```nix
-{config, ...}: {
-  ghaf.installer = {
-    enable = true;
-    imgModules = [
-      nixos-generators.nixosModules.raw-efi
-    ];
-    enabledModules = ["flushImage"];
-    installerCode = ''
-      echo "Starting flushing..."
-      if sudo dd if=${config.system.build.${config.formatAttr}} of=/dev/${config.ghaf.installer.installerModules.flushImage.providedVariables.deviceName} conv=sync bs=4K status=progress; then
-          sync
-          echo "Flushing finished successfully!"
-          echo "Now you can detach installation device and reboot to Ghaf."
-      else
-          echo "Some error occured during flushing process, exit code: $?."
-          exit
-      fi
-    '';
-  };
-}
-```
+Ghaf has a NixOS-like non-interactive, declarative installer instead of a
+conventional (imperative, non-declarative) installer as in most conventional
+Linux distributions. This is possible with the [Ghaf as
+Library](https://github.com/tiiuae/ghaf/pull/ghaf-based-project.md) approach:
+rather than clicking similar options during installation, you can configure the
+system once and deploy this configuration to the desired machines.
+
+
+To implement the pre-configured setting up approach, we used the
+[nixos-anywhere](https://github.com/nix-community/nixos-anywhere) tool.
+
+There are two separate ways to use it:
+* manually
+* with the Ghaf installation wizard.
+> NOTE: The Ghaf installation wizard is currently under development and cannot be used to create the required system configuration.
 
-After that you can build an installer image using this command:
+## Manual Installation
+
+To install Ghaf manually:
+
+1. Create your own flake using the Ghaf template:
 
 ```sh
-nix build .#nixosConfigurations.<CONFIGURATION>.config.system.build.installer
+nix flake init -t github:tiiuae/ghaf#target-x86_64-generic
 ```
 
-## Adding Installer Modules
+2. Edit it according to your preferences.
+
+3. Set the value of `ghaf.installer.sshKeys` to get an installer image.
```

Comment:
@leivos-unikie was testing this - step 3. requires more specific instructions on generating the keys and example on how to add the public key under `ghaf.installer.sshKeys`:

For example:

3.1 Generate an SSH keypair as follows:
```
$ ssh-keygen -t ed25519
Generating public/private ed25519 key pair.
Enter file in which to save the key (/home/vilvo/.ssh/id_ed25519): /home/vilvo/.ssh/id_ed25519_installer
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Your identification has been saved in /home/vilvo/.ssh/id_ed25519_installer
Your public key has been saved in /home/vilvo/.ssh/id_ed25519_installer.pub
...
```
3.2 Copy value from the public key (`id_ed25519_installer.pub` in above example) to `ghaf.installer.sshKeys` in the `flake.nix` from step 1 - for example:
```
              # Write public ssh keys that will be used to install the system.
              ghaf.installer.sshKeys = [
                "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDL9fMsclfik2U33/Kbn2CH6QjKpyAW+XOLvEH9drwI6 vilvo@carrie"
              ];
 ```

Reply:
And the line for ghaf.installer.sshKeys could already be in the template. (commented out if necessary)

Reply:
The purpose is that the keys will add SSH authorized keys that provide secure (strong key cryptography), passwordless access from your PC (installating machine) to the target device (where ghaf gets installed) over network.

For more explanation, see `nixos-anywhere` - https://github.com/nix-community/nixos-anywhere/blob/main/docs/quickstart.md#introduction